tests/repository/offer_queries_test.py

```diff
@@ -1140,12 +901,220 @@ def test_order_by_base_score(self, app):
+        assert offers[2].name == 'event_occurs_later'
+
+    @clean_database
+    def test_should_return_offers_with_varying_types(self, app):
+        # Given
+        offerer = create_offerer()
+        user = create_user()
+        venue = create_venue(offerer, postal_code='34000', departement_code='34')
+        stock1 = create_stock_with_thing_offer(offerer, venue, name='thing', thing_type=ThingType.JEUX_VIDEO)
+        stock2 = create_stock_with_thing_offer(offerer, venue, name='thing', thing_type=ThingType.CINEMA_ABO,
+                                               url='http://example.com')
+        stock3 = create_stock_with_thing_offer(offerer, venue, name='thing', thing_type=ThingType.JEUX_VIDEO)
+        stock4 = create_stock_with_thing_offer(offerer, venue, name='thing', thing_type=ThingType.JEUX_VIDEO)
+        stock5 = create_stock_with_thing_offer(offerer, venue, name='thing', thing_type=ThingType.AUDIOVISUEL)
+        stock6 = create_stock_with_thing_offer(offerer, venue, name='thing', thing_type=ThingType.JEUX)
+        create_mediation(stock1.offer)
+        create_mediation(stock2.offer)
+        create_mediation(stock3.offer)
+        create_mediation(stock4.offer)
+        create_mediation(stock5.offer)
+        create_mediation(stock6.offer)
+        PcObject.save(stock1, stock2, stock3, stock4, stock5, stock6)
+
+        def first_four_offers_have_different_type_and_onlineness(offers):
+            return len(set([o.type + (o.url or '')
+                            for o in offers[:4]])) == 4
+
+        # When
+        offers = get_active_offers(user=user, pagination_params={'seed': 0.9, 'page': 1}, departement_codes=['00'],
+                                   offer_id=None)
+        print(offers)
+
+        # Then
+        assert len(offers) == 6
+        assert first_four_offers_have_different_type_and_onlineness(offers)
+
+    @clean_database
+    def test_should_not_return_offers_with_no_stock(self, app):
+        # Given
+        product = create_product_with_thing_type(thing_name='Lire un livre', is_national=True)
+        offerer = create_offerer()
+        user = create_user()
+        venue = create_venue(offerer, postal_code='34000', departement_code='34')
+        offer = create_offer_with_thing_product(venue, product)
+        stock = create_stock_from_offer(offer, available=2, price=0)
+        user = create_user()
+        booking1 = create_booking(user, stock, venue=venue, quantity=2, is_cancelled=True)
+        booking2 = create_booking(user, stock, venue=venue, quantity=2)
+        create_mediation(stock.offer)
+        PcObject.save(booking1, booking2)
+
+        # When
+        offers = get_active_offers(user=user, pagination_params={'seed': 0.9, 'page': 1}, departement_codes=['00'],
+                                   offer_id=None)
+
+        # Then
+        assert len(offers) == 0
+
+    @clean_database
+    def test_with_no_order_by_should_return_same_number_of_recommendation(self, app):
```

Comment:
```suggestion
    def test_with_no_order_by_should_return_same_number_of_recommendations(self, app):
```
Pourquoi with_no_order ?

Reply:
J'ai retiré